net/node/transactionPool.go

```diff
@@ -46,7 +50,33 @@ func (txnPool *TXNPool) GetTxnPool(cleanPool bool) map[common.Uint256]*transacti
 	if cleanPool == true {
 		txnPool.init()
 	}
-	return list
+	return DeepCopy(list)
+}
+
+func DeepCopy(mapIn map[common.Uint256]*transaction.Transaction) map[common.Uint256]*transaction.Transaction {
+	reply := make(map[common.Uint256]*transaction.Transaction)
+	for k, v := range mapIn {
+		reply[k] = v
+	}
+	return reply
+}
+
+// Attention: clean the trasaction Pool with committed transactions.
+func (txnPool *TXNPool) CleanTxnPool(txHashes []*common.Uint256) error {
+	txsNum := len(txHashes)
+	txInPoolNum := len(txnPool.list)
+	cleaned := 0
+	for _, txHash := range txHashes {
+		if _, ok := txnPool.list[*txHash]; ok {
+			delete(txnPool.list, *txHash)
+			cleaned++
+		}
+	}
+	if len(txHashes)-cleaned != 1 {
+		log.Fatal(fmt.Sprintf("The Transactions num Unmatched. Expect %d, got %d .\n", len(txHashes), cleaned))
```

Comment:
1: the txsNum already claimed and assign the value, why not using it directly 
2:  shouldn't the txHashes equal cleaned meaning all input transactions be deleted? 
3:  follow the go fmt style, there is while space between the operation "-" eg. 
"len(txHashes) - cleaned"